test/nested_initial_state_test.dart

```diff
@@ -0,0 +1,95 @@
+@Timeout(Duration(minutes: 30))
+import 'package:fsm2/fsm2.dart';
+import 'package:fsm2/src/types.dart';
+import 'package:test/test.dart';
+
+StateMachine createMachine() {
+  final machine = StateMachine.create(
+    (g) => g
+      ..initialState<Main>()
+      ..state<Main>(
+        (g) => g
+          ..on<OnTickFirst, First>()
+          ..on<OnTickSecond, Second>()
+          ..coregion<First>(
+            (g) => g
+              ..initialState<One>()
+              ..state<One>((g) => g..on<OnToggle, Two>())
+              ..state<Two>((g) => g..on<OnToggle, One>()),
+          )
+          ..coregion<Second>(
+            (g) => g
+              ..initialState<Three>()
+              ..state<Three>((g) => g..on<OnToggle, Four>())
+              ..state<Four>((g) => g..on<OnToggle, Three>()),
+          ),
+      ),
+  );
+  return machine;
+}
+
+class Main implements State {}
+
+class First implements State {}
+
+class Second implements State {}
+
+class One implements State {}
+
+class Two implements State {}
+
+class Three implements State {}
+
+class Four implements State {}
+
+class OnTickFirst implements Event {}
+
+class OnTickSecond implements Event {}
+
+class OnToggle implements Event {}
+
+void main() {
+  test('should properly define initial state', () async {
+    final machine = createMachine();
+    expect(machine.isInState<Main>(), equals(true));
+    expect(machine.isInState<One>(), equals(false));
```

Comment:
a leaf state of a nested state machine is being set as active on boot, so this test fails here.